File: src/main/java/Logic/PetStat.java
package Logic;

public class PetStat {
    private int value;
    public static final int MIN_VALUE = 1;
    public static final int MAX_VALUE = 100;

    public PetStat(int currentValue) {
        if (currentValue > MAX_VALUE || currentValue < MIN_VALUE) {
            throw new RuntimeException(
                    "Failed to create a new pet: incorrect stats, check min and max values");
        }
        value = currentValue;
    }

    public int getValue() {
        return value;
    }

    public void increaseValue(int amount) {
        if (value + amount > MAX_VALUE) {
            value = MAX_VALUE;
        } else {
            value += amount;
        }
    }

    public void decreaseValue(int amount) {
        if (value - amount < MIN_VALUE) {
            value = MIN_VALUE;
        } else {
            value -= amount;
        }
    }
}

File: src/main/kotlin/Logic/PetStat.kt
package Logic;

class PetStat {
    private var value: Int

    companion object {
        const val MIN_VALUE: Int = 1
        const val MAX_VALUE: Int = 100
    }

    constructor() {
        value = MIN_VALUE
    }

    constructor(currentValue: Int) {
        if (currentValue > MAX_VALUE || currentValue < MIN_VALUE) {
            throw RuntimeException("Failed to create a new pet: incorrect stats, check min and max values");
        }
        value = currentValue
    }

    fun getValue(): Int {
        return value
    }

    fun increaseValue(amount: Int) {
        value = if (value + amount > MAX_VALUE) {
            MAX_VALUE
        } else {
            value + amount
        }
    }

    fun decreaseValue(amount: Int) {
        value = if (value - amount < MIN_VALUE) {
            MIN_VALUE
        } else {
            value - amount
        }
    }
}
